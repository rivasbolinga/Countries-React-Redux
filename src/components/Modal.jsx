import styled from 'styled-components';
import { useDispatch, useSelector } from 'react-redux';
import {
  closeModal, selectSelectedContinent, selectContinent, clearContinent,
} from '../redux/modalSlice';
import { filter } from '../redux/countriesSlice';

const Modal = () => {
  const selectedContinent = useSelector(selectSelectedContinent);
  const dispatch = useDispatch();
  // -- If continent selected is all, clear continent (in case there was one selected)
  // Rest of selections, we dispatch selectContinent from redux slice and close modal.
  const handleSelectContinent = (continent) => {
    if (continent === 'All') {
      dispatch(clearContinent());
      dispatch(closeModal());
    } else {
      dispatch(selectContinent(continent));
      dispatch(closeModal());
    }
    dispatch(closeModal());
    dispatch(filter());
    dispatch(closeModal());
  };

  return (
    <Wrapper>
      <div>
        <button
          type="button"
          onClick={() => {
            dispatch(closeModal());
          }}
        >
          <i className="fa-solid fa-xmark" />
        </button>
        <h2>Select Continent</h2>
        <div className="line" />
        {/* eslint-disable jsx-a11y/label-has-associated-control */}
        <div className="options">
          <div>
            <input
              type="radio"
              value="All"
              id="All"
              checked={!selectedContinent}
              onChange={() => handleSelectContinent('All')}
            />
            <label htmlFor="All">All</label>
          </div>
          <div>
            <input
              type="radio"
              value="Africa"
              id="Africa"
              checked={selectedContinent === 'Africa'}
              onChange={() => handleSelectContinent('Africa')}
            />
            <label htmlFor="Africa">Africa</label>
          </div>
          <div>
            <input
              type="radio"
              value="Asia"
              id="Asia"
              checked={selectedContinent === 'Asia'}
              onChange={() => handleSelectContinent('Asia')}
            />
            <label htmlFor="Asia">Asia</label>
          </div>
          <div>
            <input
              type="radio"
              value="Europe"
              id="Europe"
              checked={selectedContinent === 'Europe'}
              onChange={() => handleSelectContinent('Europe')}
            />
            <label htmlFor="Europe">Europe</label>
          </div>
          <div>
            <input
              type="radio"
              value="North America"
              id="North America"
              checked={selectedContinent === 'North America'}
              onChange={() => handleSelectContinent('North America')}
            />
            <label htmlFor="North America">North America</label>
          </div>
          <div>
            <input
              type="radio"
              value="South America"
              id="South America"
              checked={selectedContinent === 'South America'}
              onChange={() => handleSelectContinent('South America')}
            />
            <label htmlFor="South America">South America</label>
          </div>
          <div>
            <input
              type="radio"
              value="Oceania"
              id="Oceania"
              checked={selectedContinent === 'Oceania'}
              onChange={() => handleSelectContinent('Oceania')}
            />
            <label htmlFor="Oceania">Oceania</label>
          </div>
          <div>
            <input
              type="radio"
              value="Antarctica"
              id="Antarctica"
              checked={selectedContinent === 'Antarctica'}
              onChange={() => handleSelectContinent('Antarctica')}
            />
            <label htmlFor="Antarctica">Antarctica</label>
          </div>
        </div>
      </div>
    </Wrapper>
  );
};
const Wrapper = styled.section`
  width: 92%;
  height: 325px;
  border-radius: 10px;
  border: 4px solid white;
  background-color: white;
  color: white;
  position: absolute;
  top: 50%;
  left: 50%;
  background-color: #535bf2;
  padding: 10px;
  background-image: url('https://upload.wikimedia.org/wikipedia/commons/thumb/e/ec/World_Map_Blank.svg/1600px-World_Map_Blank.svg.png');
  background-size: 60%;
  background-repeat: no-repeat;
  background-position: 130px;
  transform: translate(-50%, -50%);
   button {
    color: white;
    background: none;
  }
  .line {
    height: 2px;
    background-color: white;
  }
  .options {
    margin-top: 10px;
  }
  @media (min-width:768px) {
    width: 400px;
  }
`;

export default Modal;
