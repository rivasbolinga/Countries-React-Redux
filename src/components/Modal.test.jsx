import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store'; // installed as a dev dependency
import Modal from './Modal';

const mockStore = configureStore([]);
const store = mockStore({
  continents: {
    selectedContinent: null,
    allContinents: [
      { id: 1, name: 'Africa' },
      { id: 2, name: 'Antarctica' },
      { id: 3, name: 'Asia' },
      { id: 4, name: 'Australia' },
      { id: 5, name: 'Europe' },
      { id: 6, name: 'North America' },
      { id: 7, name: 'South America' },
    ],
  },
});

describe('Modal component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Modal />
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
