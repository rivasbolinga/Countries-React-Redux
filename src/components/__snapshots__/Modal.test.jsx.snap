// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal component > should render correctly 1`] = `
<section
  className="sc-beqWaB gELQXw"
>
  <div>
    <button
      onClick={[Function]}
      type="button"
    >
      <i
        className="fa-solid fa-xmark"
      />
    </button>
    <h2>
      Select Continent
    </h2>
    <div
      className="line"
    />
    <div
      className="options"
    >
      <div>
        <input
          checked={true}
          id="All"
          onChange={[Function]}
          type="radio"
          value="All"
        />
        <label
          htmlFor="All"
        >
          All
        </label>
      </div>
      <div>
        <input
          checked={false}
          id="Africa"
          onChange={[Function]}
          type="radio"
          value="Africa"
        />
        <label
          htmlFor="Africa"
        >
          Africa
        </label>
      </div>
      <div>
        <input
          checked={false}
          id="Asia"
          onChange={[Function]}
          type="radio"
          value="Asia"
        />
        <label
          htmlFor="Asia"
        >
          Asia
        </label>
      </div>
      <div>
        <input
          checked={false}
          id="Europe"
          onChange={[Function]}
          type="radio"
          value="Europe"
        />
        <label
          htmlFor="Europe"
        >
          Europe
        </label>
      </div>
      <div>
        <input
          checked={false}
          id="North America"
          onChange={[Function]}
          type="radio"
          value="North America"
        />
        <label
          htmlFor="North America"
        >
          North America
        </label>
      </div>
      <div>
        <input
          checked={false}
          id="South America"
          onChange={[Function]}
          type="radio"
          value="South America"
        />
        <label
          htmlFor="South America"
        >
          South America
        </label>
      </div>
      <div>
        <input
          checked={false}
          id="Oceania"
          onChange={[Function]}
          type="radio"
          value="Oceania"
        />
        <label
          htmlFor="Oceania"
        >
          Oceania
        </label>
      </div>
      <div>
        <input
          checked={false}
          id="Antarctica"
          onChange={[Function]}
          type="radio"
          value="Antarctica"
        />
        <label
          htmlFor="Antarctica"
        >
          Antarctica
        </label>
      </div>
    </div>
  </div>
</section>
`;
